sdk/storage/azblob/container/client_test.go

```diff
@@ -2222,3 +2224,786 @@ func (s *ContainerUnrecordedTestsSuite) TestSASContainerClient() {
+	for _, subResp := range resp.Responses {
+		_require.NotNil(subResp.Response)
+		if subResp.Response.StatusCode >= 200 && subResp.Response.StatusCode < 300 {
+			ctr++
+		}
+	}
+	_require.Equal(ctr, 10)
+
+	pager = containerClient.NewListBlobsFlatPager(nil)
+	ctr = 0
+	for pager.More() {
+		resp, err := pager.NextPage(context.Background())
+		handleError(err)
+		ctr += len(resp.ListBlobsFlatSegmentResponse.Segment.BlobItems)
+	}
+	_require.Equal(ctr, 0)
+}
+
+func (s *ContainerUnrecordedTestsSuite) TestContainerBlobBatchSetTierPartialFailureUsingSharedKey() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerName := testcommon.GenerateContainerName(testName)
+	containerClient := testcommon.CreateNewContainer(context.Background(), _require, containerName, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	bb, err := containerClient.NewBatchBuilder()
+	_require.NoError(err)
+
+	// add 5 blobs to BatchBuilder which does not exist
+	for i := 0; i < 15; i++ {
+		bbName := fmt.Sprintf("blockblob%v", i)
+		if i < 10 {
+			_ = testcommon.CreateNewBlockBlob(context.Background(), _require, bbName, containerClient)
+		}
+		err = bb.SetTier(bbName, blob.AccessTierCool, nil)
+		_require.NoError(err)
+	}
+
+	pager := containerClient.NewListBlobsFlatPager(nil)
+	var ctrHot, ctrCool = 0, 0
+	for pager.More() {
+		resp, err := pager.NextPage(context.Background())
+		handleError(err)
+		for _, blobItem := range resp.ListBlobsFlatSegmentResponse.Segment.BlobItems {
+			if *blobItem.Properties.AccessTier == container.AccessTierHot {
+				ctrHot++
+			} else if *blobItem.Properties.AccessTier == container.AccessTierCool {
+				ctrCool++
+			}
+		}
+	}
+	_require.Equal(ctrHot, 10)
+	_require.Equal(ctrCool, 0)
+
+	resp, err := containerClient.SubmitBatch(context.Background(), bb, nil)
+	_require.Error(err)
+	_require.NotEmpty(resp.RequestID)
+
+	var ctrSuccess, ctrFailure = 0, 0
+	for _, subResp := range resp.Responses {
+		_require.NotNil(subResp.Response)
+		if subResp.Response.StatusCode >= 200 && subResp.Response.StatusCode < 300 {
+			ctrSuccess++
+		} else {
+			ctrFailure++
+		}
+	}
+	_require.Equal(ctrSuccess, 10)
+	_require.Equal(ctrFailure, 5)
+
+	pager = containerClient.NewListBlobsFlatPager(nil)
+	ctrHot = 0
+	ctrCool = 0
+	for pager.More() {
+		resp, err := pager.NextPage(context.Background())
+		handleError(err)
+		for _, blobItem := range resp.ListBlobsFlatSegmentResponse.Segment.BlobItems {
+			if *blobItem.Properties.AccessTier == container.AccessTierHot {
+				ctrHot++
+			} else if *blobItem.Properties.AccessTier == container.AccessTierCool {
+				ctrCool++
+			}
+		}
+	}
+	_require.Equal(ctrHot, 0)
+	_require.Equal(ctrCool, 10)
+}
+
+/*
+func (s *ContainerUnrecordedTestsSuite) TestContainerBlobBatchDeleteUsingTokenCredential() {
+	_require := require.New(s.T())
```

Comment:
Is there a reason for leaving token credential tests out?

Reply:
I commented this part because we currently don't have the OAuth credentials in our DevOps pipeline.
Will uncomment these tests once it is added.